guidance_pipeline/read_data.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"encoding/csv"
+	"os"
+)
+
+func readData(file string) ([][]string, error) {
```

Comment:
Update to use a `Reader` interface and pass the file in. That will let opening the file bubble up to main and make it clear what the resources being used are.